s3-handler/main.go

```diff
@@ -108,6 +111,24 @@ func Handler(request events.S3Event) (Response, error) {
 				parseXRayTraceID(tmp.(string), parsedLine)
 			}
 
+			if req, ok := parsedLine["request"].(string); ok {
+				if redactPattern != nil {
+					parsedLine["request"] = redactRequest(req, redactPattern)
+				}
+
+				if parserType == "alb" {
+					httpMeta, err := parseRequestHttpMeta(req)
```

Comment:
If redaction happens, it feels like parsing the request should be done with the redacted version of the string.